dgp/utils/render_3d_to_2d.py

```diff
@@ -0,0 +1,422 @@
+    ontology_name_mapper: dict
+        Map the class names from bounding_box_2d to bounding_box_3d if the class names are different.
+        eg: {'Pedestrian': 'Person','Car': 'Car'}.
+    camera_datum_names: Optional[list[str]]
+        List of camera names.
+        If None, use all the cameras available in the DGP dataset.
+    lidar_datum_names: Optional[list[str]]
+        List of lidar names.
+        If None, use all the lidars available in the DGP dataset.
+    max_num_items: Optional[int]
+        If not None, then show only up to this number of items. This is useful for debugging a large dataset.
+        Default: None.
+    Returns
+    -------
+    associated_bboxes: defaultdict[defaultdict[list[Tuple]]]
+        a dictionary with key is the camera name, value is a dictionary whose key is class_name of bounding_box_2d,
+        value is list of Tuple [bounding_box_2d, bounding_box_3d, centroid_2d]
+    """
+    datum_names = camera_datum_names + lidar_datum_names
+    dataset = SynchronizedSceneDataset(
+        scenes_dataset_json,
+        datum_names=datum_names,
+        requested_annotations=[ANNOTATIONS_2D, ANNOTATIONS_3D],
+        only_annotated_datums=True
+    )
+    associated_bboxes = associate_3d_and_2d_annotations(
+        dataset, ontology_name_mapper, camera_datum_names, lidar_datum_names, max_num_items
+    )
+    return associated_bboxes
+
+
+def draw_bounding_box_2d_distance_distribution(
+    scenes_dataset_json: str, ontology_name_mapper: dict, output_dir: str, camera_datum_names: Optional[list[str]],
+    lidar_datum_names: Optional[list[str]], max_num_items: Optional[int]
+):
+    """Draw the distance's distributution histogram of 2d bounding boxes by associating bounding_box_3d of the same object.
+    Parameters
+    ----------
+    scenes_dataset_json: str
+        Full path to the dataset scene json.
+    ontology_name_mapper: dict
+        Map the class names from bounding_box_2d to bounding_box_3d if the class names are different.
+        eg: {'Pedestrian': 'Person','Car': 'Car'}.
+    output_dir: str
+        Path to save the histogram picture.
+    camera_datum_names: Optional[list[str]]
+        List of camera names.
+        If None, use all the cameras available in the DGP dataset.
+    lidar_datum_names: Optional[list[str]]
+        List of lidar names.
+        If None, use all the lidars available in the DGP dataset.
+    max_num_items: Optional[int]
+        If not None, then show only up to this number of items. This is useful for debugging a large dataset.
+        Default: None.
+    """
+    associated_bboxes = associate_3d_and_2d_annotations_dataset(
+        scenes_dataset_json, ontology_name_mapper, camera_datum_names, lidar_datum_names, max_num_items
+    )
+    os.makedirs(output_dir, exist_ok=True)
+    # Summarize statistics per camera per class over all scenes.
+    for camera_name in camera_datum_names:
+        for name_2d, name_3d in ontology_name_mapper.items():
```

Comment:
nit: we have an ontology mapper class https://github.com/TRI-ML/dgp/blob/master/dgp/annotations/transforms.py#L90 that could handle converting all annotations to the same ontology.